pybaseball/datahelpers/caching/dataframe_utils.py

```diff
@@ -0,0 +1,58 @@
+import pandas as pd
+
+from .cache_type import CacheType
+from typing import Optional
+
+
+def load(filename: str, cache_type: CacheType) -> pd.DataFrame:
+    if cache_type in [CacheType.CSV, CacheType.CSV_BZ2, CacheType.CSV_GZ, CacheType.CSV_XZ, CacheType.CSV_ZIP]:
+        data = pd.read_csv(filename)
+    elif cache_type in [CacheType.FEATHER, CacheType.FEATHER_LZ4, CacheType.FEATHER_UNCOMPRESSED, CacheType.FEATHER_ZSTD]:
+        data = pd.read_feather(filename)
+    elif cache_type in [CacheType.JSON, CacheType.JSON_BZ2, CacheType.JSON_GZ, CacheType.JSON_XZ, CacheType.JSON_ZIP]:
+        data = pd.read_json(filename)
+    elif cache_type in [CacheType.PARQUET, CacheType.PARQUET_GZ, CacheType.PARQUET_FAST, CacheType.PARQUET_FAST_GZ]:
+        engine = 'pyarrow'
+        if cache_type in [CacheType.PARQUET_FAST, CacheType.PARQUET_FAST_GZ]:
+            engine = 'fastparquet'
+        data = pd.read_parquet(filename, engine=engine)
+    elif cache_type in [CacheType.PICKLE, CacheType.PICKLE_BZIP, CacheType.PICKLE_GZ, CacheType.PICKLE_XZ, CacheType.PICKLE_ZIP]:
+        data = pd.read_pickle(filename)
+    else:
+        raise ValueError(f"cache_type of {cache_type} is unsupported.")
+    data.drop(
+        data.columns[data.columns.str.contains('unnamed', case=False)],
+        axis=1,
+        inplace=True
+    )
+    return data
```

Comment:
Snoop it off the extension. I think:
```python3
ext_to_func = {'csv' : pd.read_csv, 'pickle': pd.read_pickle, 'parquet': pd.read_parquet}
def load(filename):
  ext = filename.split('.')[-1]
  return ext_to_func[ext](filename)
```

Reply:
That could be possible once we decide on the final caching options are. I'd like to wait for other input to make sure we make the right decision there.